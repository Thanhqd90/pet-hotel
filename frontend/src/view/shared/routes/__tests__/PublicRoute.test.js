import { mount } from 'enzyme';
import React, { Component } from 'react';
import {
  BrowserRouter,
  Link,
  Route,
  Switch,
} from 'react-router-dom';
import PublicRoute from 'view/shared/routes/PublicRoute';

function mountPage(currentUser) {
  const PublicPage = class PublicPage extends Component {
    render() {
      return <div data-test="PublicPage">Public Page</div>;
    }
  };

  return mount(
    <BrowserRouter>
      <div>
        <Link to="/public">Go to public</Link>

        <Switch>
          <Route
            exact
            path="/"
            render={() => (
              <div data-test="PrivatePage">Private</div>
            )}
          />

          <PublicRoute
            path="/public"
            currentUser={currentUser}
            component={PublicPage}
          />
        </Switch>
      </div>
    </BrowserRouter>,
  );
}

describe('view/shared/routes/PublicRoute', () => {
  describe('Authenticated', () => {
    let component;
    beforeAll(() => {
      component = mountPage({ id: 1 });

      component
        .find('Link[to="/public"]')
        .simulate('click', { button: 0 });
    });

    test('redirect to private', () => {
      expect(component.html()).toMatchSnapshot();

      expect(
        component
          .find(`[data-test="PrivatePage"]`)
          .exists(),
      ).toBeTruthy();
    });
  });

  describe('Not authenticated', () => {
    let component;

    beforeAll(() => {
      component = mountPage(null);

      component
        .find('Link[to="/public"]')
        .simulate('click', { button: 0 });
    });

    test('opens the page', () => {
      expect(component.html()).toMatchSnapshot();

      expect(
        component.find(`[data-test="PublicPage"]`).exists(),
      ).toBeTruthy();
    });
  });
});
