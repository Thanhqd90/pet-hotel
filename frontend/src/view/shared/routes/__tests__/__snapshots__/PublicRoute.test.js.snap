// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view/shared/routes/PublicRoute Authenticated redirect to private 1`] = `"<div><a href=\\"/public\\">Go to public</a><div data-test=\\"PrivatePage\\">Private</div></div>"`;

exports[`view/shared/routes/PublicRoute Not authenticated opens the page 1`] = `"<div><a href=\\"/public\\">Go to public</a><div data-test=\\"PublicPage\\">Public Page</div></div>"`;
